src/Adapter/Filesystem/Satellite.php

```diff
@@ -45,6 +44,10 @@ public function build(
         foreach ($this->files as $file) {
             if ($file instanceof Packaging\DirectoryInterface) {
                 foreach (new \RecursiveIteratorIterator($file) as $current) {
+                    $dir = $this->workdir . '/' . substr($current->getPath(), 0, strrpos($current->getPath(), '/'));
```

Comment:
Use the [`basename`](https://www.php.net/manual/fr/function.basename.php) or [`dirname`](https://www.php.net/manual/fr/function.dirname.php) function to do this.